omod/src/main/webapp/resources/htmlforms/Walkin-Client-Data-Capture-tool.xml

```diff
@@ -0,0 +1,256 @@
+<htmlform formName="Walk-in ART Service" formDescription="Walk-in Client Data Capture Tool" formEncounterType="d18bd8f2-dfcd-11eb-ba80-0242ac130004" formUuid="aa858f72-0080-44a8-8bac-55bc275e1d10" formVersion="3.0" formAddMetadata="yes" formUILocation="patientDashboard.visitActions" formOrder="3" formIcon="icon-medkit" formShowIf="(visit.active || !visit.active) &amp;&amp; patient.person.dead==false &amp;&amp; (sessionLocation.uuid=='86863db4-6101-4ecf-9a86-5e716d6504e4' || sessionLocation.uuid=='7c231e1a-1db5-11ea-978f-2e728ce88125')" formDisplayStyle="Standard">
+
+    <script type="text/javascript">
+
+            if (jQuery) {
+    
+                jq(document).ready(function () {   
+                    
+                    //Block Multiple encounter entry on same date
+                    blockEncounterOnSameDateEncounter(jq("#encounterDate").find("input[type=hidden]"),"block");
+    
+                    disable_fields("other-regimen");                    
+                    disable_fields("reason-for-visit-other")                    
+
+                    jq("#art-regimen").change(function () {
+                        if (jq(this).find("select").find(":selected").val() == "90002") {
+                            enable_fields("other-regimen");
+                        } else {
+                            disable_fields("other-regimen");
+                        }
+                    });                 
+
+                    jq("#reason-for-visit").change(function () {
+                        if (jq(this).find("select").find(":selected").val() == "90002") {
+                            enable_fields("reason-for-visit-other");
+                        } else {
+                            disable_fields("reason-for-visit-other");
+                        }
+                    });
+    
+                    beforeSubmit.push(function () {
+                        var submitVal = true;
+
+                        var submitValStage = [];
+                        var submitValStageART = [];
+
+                        if(getValue('telephone-number.value')==""){
```

Comment:
@ODORA0 Instead of the Javascript for the required fields, use the required values for each of the Obs widgets, I think you can add custom error messages too

Reply:
@ssmusoke Please clarify here. Something like this?

> if (getValue('isoniazid-trigger.value') != "" &amp;&amp; getValue('isoniazid-pills.value') == "") {
    getField('isoniazid-pills.error').html('No of isoniazid pills required').show();
    return false;
}

Reply:
@ODORA0 Check the Obs tag it has a required attribute

Reply:
@ODORA0 was this fixed

Reply:
@ssmusoke I will remove this Javascript to avoid confusion, the Patient Attribute tag tag doesn't exist for now as well anyway. So we can't mark the phone number as required since we are not using `<obs id="phone-number" conceptId="90266" />`

Reply:
@ODORA0 Just remove the phone number for now